datashuttle/utils/folders.py

```diff
@@ -463,3 +470,91 @@ def search_filesystem_path_for_folders(
         else:
             all_filenames.append(os.path.basename(file_or_folder))
     return all_folder_names, all_filenames
+
+
+# -----------------------------------------------------------------------------
+# Get Next Subject or Session Number
+# -----------------------------------------------------------------------------
+
+
+def get_next_sub_or_ses_number(
+    cfg: Configs, sub: Optional[str], search_str: str
+) -> Tuple[int, int]:
+    """
+    Suggest the next available subject or session number. This function will
+    search the local repository, and the remote repository, for all subject
+    or session folders (subject or session depending on inputs).
+    It will take the union of all folder names, find the relevant key-value
+    pair values, and return the maximum value + 1 as the new number.
+    A warning will be shown if the existing sub / session numbers are not
+    consecutive.
+
+    If sub is None, the top-level level folder will be searched (i.e. for subjects).
+    The search string "sub-*" is suggested in this case. Otherwise, the subject,
+    level folder for the specified subject will be searched. The search_str
+    "ses-*" is suggested in this case.
+
+    Parameters
+    ----------
+
+    cfg : datashuttle configs class
+    sub : subject name to search within if searching for sessions, otherwise None
+          to search for subjects
+    search_str : the string to search for within the top-level or subject-levle
+                 folder ("sub-*") or ("ses-*") are suggested, respectively.
+
+    Returns
+    -------
+    suggested_new_num : the new suggested sub / ses number.
+    latest_existing_num : the latest sub / ses number that currents
+                          exists in the project.
+    Notes
+    -----
+    Should this search filenames also?
+
+    """
+    if sub:
+        bids_key = "ses"
+    else:
+        bids_key = "sub"
+
+    # Search local and remote for folders that begin with "sub-*"
+    (local_foldernames, __,) = search_sub_or_ses_level(
```

Comment:
In here and line 529, is there a reason to use a double underscore? I always used a single to signify an unused variable. Does `__` mean something different?

Also, are the parentheses needed?

Reply:
Funnily enough I used doubles based on some SO answers I read early when learning to code such as [this ](https://softwareengineering.stackexchange.com/a/285304), that indicate `_` can conflict with `gettext`. Having no idea what `gettext` was, I used `__` by default to be safe. However it appears this is unecessary and `_` is standard (unless your specifically using `gettext` in the codebase. Will change across the codebase in a new PR.

Not sure about the brackets I think this is some `black` magic, I'll check.